securedrop/models.py

```diff
@@ -317,12 +317,12 @@ def regenerate_totp_shared_secret(self):
         self.otp_secret = pyotp.random_base32()
 
     def set_hotp_secret(self, otp_secret):
-        self.is_totp = False
         self.otp_secret = base64.b32encode(
             binascii.unhexlify(
                 otp_secret.replace(
                     " ",
                     "")))
+        self.is_totp = False
```

Comment:
Can you document why you are doing this in the commit message? 

Note that the validation of the secret value itself should be done prior to `set_hotp_secret` being called - i.e. in `manage._add_user` and in the 2FA-related views and/or forms (the exception being only when we do some test setup where we might not run the validation).